plugins/global-search/src/components/Results.tsx

```diff
@@ -1,108 +1,192 @@
-import { useMemo } from "react"
-import { assertNever } from "../utils/assert"
-import type { EntryResult } from "../utils/filter/group-results"
+import { defaultRangeExtractor, useVirtualizer } from "@tanstack/react-virtual"
+import { type CSSProperties, useCallback, useMemo, useRef, useState } from "react"
+import { cn } from "../utils/className"
+import type { PreparedGroup, PreparedResult } from "../utils/filter/group-results"
 import { ResultType } from "../utils/filter/types"
-import type { RootNodeType } from "../utils/indexer/types"
-import { useFocusHandlers } from "../utils/useFocus"
+import { GroupHeader } from "./GroupHeader"
 import { Match } from "./Match"
-import { IconArrowRight } from "./ui/IconArrowRight"
-import { IconCode } from "./ui/IconCode"
-import { IconCollection } from "./ui/IconCollection"
-import { IconComponent } from "./ui/IconComponent"
-import { IconWebPage } from "./ui/IconWebPage"
 
 interface ResultsProps {
-    groupedResults: readonly EntryResult[]
+    readonly groups: readonly PreparedGroup[]
 }
 
-export function ResultsList({ groupedResults }: ResultsProps) {
+export function ResultsList({ groups }: ResultsProps) {
+    const scrollElementRef = useRef<HTMLDivElement>(null)
+    const activeStickyIndexRef = useRef(0)
+    const { collapsedGroups, toggleGroup } = useCollapsedGroups()
+    const { virtualItems, stickyIndexes } = useProcessedResults(groups, collapsedGroups)
+
+    const isActiveSticky = useCallback((index: number) => activeStickyIndexRef.current === index, [])
+
+    const rowVirtualizer = useVirtualizer({
+        overscan: 10,
```

Comment:
Did you arrive at this through experiments? Asking since the default is `1`.

Reply:
I found it "feeling" better when scrolling a bit faster – no science or real numbers were harmed in this experiment.